daphne_worker/src/config.rs

```diff
@@ -818,3 +818,54 @@ pub(crate) enum DaphneWorkerDeployment {
     /// internal test API.
     Dev,
 }
+
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub(crate) struct HpkeReceiverKvKey {
+    pub(crate) version: DapVersion,
+    pub(crate) hpke_config_id: u8,
+}
+
+impl HpkeReceiverKvKey {
+    fn parse_from_name(name: &str) -> Option<Self> {
```

Comment:
This parser accepts invalid input, namely "trailing junk", e.g.

hpke_receiver_config/version/v03/config_id/123/this/is/junk

would be accepted because we don't do

if iter.next().is_some() {
    return None;
}

after parsing the id.  This doesn't actually break anything, but it's better to only accept valid.